unit-tests/src/test/scala/scala/HashCodeTest.scala

```diff
@@ -2,15 +2,29 @@ package scala
 
 import org.junit.Test
 import org.junit.Assert._
+import scala.scalanative.buildinfo.ScalaNativeBuildInfo
 
 class HashCodeTest {
   case class MyData(string: String, num: Int)
 
+  def scala212orOlder: Boolean =
+    ScalaNativeBuildInfo.scalaVersion
+      .split('.')
+      .take(3)
+      .map(_.toInt) match {
+      case Array(2, n, _) if n <= 12 => true
+      case _                         => false
+    }
```

Comment:
That won't work for milestones, snapshots and RCs, which is going to be very annoying. Consider the following instead:
```suggestion
  def scala212orOlder: Boolean = {
    ScalaNativeBuildInfo.scalaVersion.startsWith("2.11.") ||
      ScalaNativeBuildInfo.scalaVersion.startsWith("2.12.")
  }
```